### PR TITLE
Add XML schema files

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
I was able to find the original XML schema files and it's actually super robust. By using this I'll be able to limit the python validation I'll have to do

- Adds `schema/` folder with the schema files I found on Oscar
- Re-organizes `caveschema.xsd` and adds some comments
- Fixes key/keyref pairs in the schema files (scope issue)
- Adds all 5 walls to `<PlacementRoot>` in the test project
- Adds `lxml` package to the project

Closes issue #29 
Closes issue #24

### CODE REVIEW

File: schema/caveschema.xsd [hunk 1]
```diff
@@ -0,0 +1,1049 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (broarr) [inline schema/caveschema.xsd:1]: Nice, this thing is it!

File: schema/caveschema.xui [hunk 2]
```diff
@@ -0,0 +1,1355 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?jaxfront version=2.2;time=2006-12-22 05:29:11.631;include=types.xui;xui=file:/C:/Documents%20and%20Settings/Ilya%20Kreymer/My%20Documents/workspace/CaveUI/schema/brandnew.xui;xsd=brandnew.xsd;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui;include=types.xui?>
+<XUI xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.0">
```
Comment (broarr) [inline schema/caveschema.xui:3]: This is Android UI interface?

**Comment by RobertGemmaJr** [comment]
> Were there different versions of cave writing or something?

Yes, it certainly looks that way. `caveschema.xsd` asserts that `<Story version="8" /> and (I'm just now realizing) there are 7 `xslt` files.  I'll be able to use them to convert any older projects up to version 8 - at least in theory haha

File: schema/caveschema.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (RobertGemmaJr) [reply schema/caveschema.xml:1]: I'll delete

**Pull request merged**